### PR TITLE
Port and cross compile test for java.util.Collections.

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Jenkins, retest this please.

File: test-suite/js/src/test/scala/org/scalajs/testsuite/javalib/util/CollectionsJSTest.scala [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+import org.scalajs.jasminetest.JasmineTest
+import org.scalajs.testsuite.utils.AssertThrows._
+
+object CollectionsJSTest extends JasmineTest with CollectionsTestBase {
+
+  def testCheckedCollection[D, E <: D](newColl: () => ju.Collection[E], elem: E,
+      wrongElem: D): Unit = {
+    def superColl(): ju.Collection[D] = newColl().asInstanceOf[ju.Collection[D]]
+    it("should add instances of the correct type (Collection methods)") {
```
Comment (sjrd) [inline test-suite/js/src/test/scala/org/scalajs/testsuite/javalib/util/CollectionsJSTest.scala]: Add a blank line above this line.

File: test-suite/js/src/test/scala/org/scalajs/testsuite/javalib/util/CollectionsJSTest.scala [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+          singletonMap().entrySet().head.setValue(wrongValue))
+    }
+  }
+
+  def testCheckedSortedMap[J, K <: J, U, V <: U](
+      newSortedMap: () => ju.SortedMap[K, V], key: K, wrongKey: J, value: V,
+      wrongValue: U, recursive: Boolean = false): Unit = {
+    def superSortedMap(): ju.SortedMap[J, U] =
+      newSortedMap().asInstanceOf[ju.SortedMap[J, U]]
+    def testSubsortedmap(fun: ju.SortedMap[K, V] => ju.SortedMap[K, V]): Unit = {
```
Comment (sjrd) [inline test-suite/js/src/test/scala/org/scalajs/testsuite/javalib/util/CollectionsJSTest.scala]: Blank line above this line.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/CollectionsOnSynchronizedListTest.scala [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    extends CollectionsSynchronizedListTest {
+  def originalFactory: ListFactory = new LinkedListFactory
+}
+
+class CollectionsOnSynchronizedListCopyOnWriteArrayListTest
+    extends CollectionsSynchronizedListTest {
+  def originalFactory: ListFactory = new CopyOnWriteArrayListFactory
+}
+
+
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/CollectionsOnSynchronizedListTest.scala]: 2 EOLs too many.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/CollectionsOnMapsTest.scala [hunk 4]
```diff
@@ -0,0 +1,73 @@
+package org.scalajs.testsuite.javalib.util
+
+import java.{lang => jl, util => ju}
+
+import org.junit.Test
+import org.scalajs.testsuite.utils.CollectionsTestBase
+
+
+trait CollectionsOnMapsTest extends CollectionsTestBase {
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/CollectionsOnMapsTest.scala]: Double blank line.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/CollectionsOnSynchronizedSetTest.scala [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2015, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.javalib.util
+
+import java.{util => ju}
+
+
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/CollectionsOnSynchronizedSetTest.scala]: Double blank line.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/LinkedHashMapTest.scala [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+
+    assertEquals(expectedSize, lhm.keySet.size)
+    for ((key, index) <- lhm.keySet.zipWithIndex)
+      assertEquals(expectedKey(index), key)
+
+    assertEquals(expectedSize, lhm.entrySet.size)
+    for ((value, index) <- lhm.values.zipWithIndex)
+      assertEquals(expectedValue(index), value)
+
+  }
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/LinkedHashMapTest.scala]: Spurious blank line.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/LinkedHashMapTest.scala [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/LinkedHashMapTest.scala]: Spurious blank line.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/LinkedHashMapTest.scala [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+    for ((value, index) <- lhm.values.zipWithIndex)
+      assertEquals(expectedValue(index), value)
+  }
+
+}
+
+object LinkedHashMapFactory {
+  def allFactories: Iterator[MapFactory] = {
+    Iterator(new LinkedHashMapFactory(true, true), new LinkedHashMapFactory(true, false),
+      new LinkedHashMapFactory(false, true), new LinkedHashMapFactory(false, false))
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/LinkedHashMapTest.scala]: 4 spaces

**Comment by sjrd** [comment]
That's all. So just a couple minor things.

**Comment by nicolasstucki** [comment]
Updated

**Comment by sjrd** [comment]
LGTM

**Comment by sjrd** [comment]
retest this please

**Pull request merged**